src/test/java/com/revature/controllers/QuestionControllerTests.java

```diff
@@ -5,58 +5,88 @@
 import static org.junit.Assert.assertNotEquals;
 import static org.junit.Assert.assertTrue;
 import static org.mockito.Mockito.when;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
-import java.time.LocalDate;
+
 import java.util.ArrayList;
 import java.util.List;
 
-import org.junit.Ignore;
+import java.sql.*;	
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.Mockito;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+
+import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.PageImpl;
-import org.springframework.data.domain.PageRequest;
 import org.springframework.data.domain.Pageable;
 import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+
 import org.springframework.test.context.junit4.SpringRunner;
 import org.springframework.test.web.servlet.MockMvc;
 import org.springframework.test.web.servlet.MvcResult;
 import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import com.revature.Application;
 
-import com.revature.controller.QuestionController;
-import com.revature.entities.Answer;
 import com.revature.entities.Question;
+import com.revature.entities.User;
 import com.revature.services.QuestionService;
 
 
 @RunWith(SpringRunner.class)
-@WebMvcTest(QuestionController.class)
+@SpringBootTest(
+		webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+		classes = Application.class)
+@AutoConfigureMockMvc
 public class QuestionControllerTests {
-
-    @Autowired 
-    private ObjectMapper mapper;
     
+	@Autowired 
+	private ObjectMapper mapper;
+	
+	@Autowired
+	private WebApplicationContext context;
+
+	static User u1;
+	
 	@Autowired
 	private MockMvc mvc;
 	
+
 	@MockBean
 	private QuestionService questionService;
 	
+	private Date ts = new Date(100000);
```

Comment:
We should not be using `java.sql.Date` unless you are conducting integration tests with the Database that necessitate using it. Use LocalDate for all internal application date handling.

Reply:
fixed